apps/etl-func/src/func.ts

```diff
@@ -34,6 +60,43 @@ const main = async (config: Config) => {
     methods: ["GET"],
     route: "health",
   });
+
+  const messageIngestion = async (documents: unknown[]) => {
+    const messagesToSend: MessageEvent[] = [];
+    for (const messageMetadataFromCosmosDB of documents) {
+      const messageMetadata = messageMetadataSchema.parse(
+        messageMetadataFromCosmosDB,
+      );
+
+      const message =
+        await messageAdapter.getMessageByMetadata(messageMetadata);
+
+      if (message !== undefined) {
+        const messageEvent = await getMessageEventFromMessage(
+          message,
+          PDVTokenizer,
+        );
+        messagesToSend.push(messageEvent);
+      } else {
+        logger.info(
+          `${messageMetadata.id} returned an undefined message event`,
+        );
+      }
+    }
+
+    await producer.publish(messagesToSend);
+  };
+
+  app.cosmosDB("messagesCosmosDBTrigger", {
+    connection: "COSMOSDB_URI",
+    containerName: config.messageCosmosDB.containerName,
+    createLeaseContainerIfNotExists: true,
+    databaseName: config.messageCosmosDB.databaseName,
+    handler: messageIngestion,
+    leaseContainerName: `${config.messageCosmosDB.containerName}-ingestion-lease`,
```

Comment:
the defualt name is **lease**.  There is a a high possibility that lease container is already used by other services in production than i've decided to divide them at least for this test scenario.

Reply:
you should use `leaseContainerPrefix` instead (https://learn.microsoft.com/en-us/azure/azure-functions/functions-bindings-cosmosdb-v2-trigger?tabs=python-v2%2Cisolated-process%2Cextensionv4%2Cnodejs-v4&pivots=programming-language-javascript#configuration)

setting a prefix enables us to reuse the `lease` container if it exists

Reply:
Yes we can try like that but in the io-p-comsos-api production dbs there is no **lease** named container and it seems that each service is creating is own one.

so what we can do is:

1. create a new one test ingestion lease like i'm doing. Creating a new one i think is more safe for testing purpose and we can delete it once the test is finished(i'm going to create it with terraform)
2. you can indicate to me which of the production lease container we must share with other services/functions and so we can use leaseContainerPrefix. In this last case we need to be able to delete the leases items for do a test retry if something goes wrong with the function execution (it can be a blocking and temporary error with PDV for example )and i think this is too risky in the prod enviroment with a shared container.

what do you think ?